ts/Dashboards/EditMode/Menu/MenuItem.ts

```diff
@@ -133,25 +112,32 @@ class MenuItem {
             return;
         }
 
-        return renderItem(container, this.getElementOptions(options, callback));
+        const value = this.options.getValue && this.options.getValue(item);
+        const callback = function (): void {
+            if (options.events && options.events.click) {
+                options.events.click.apply(item, arguments);
+            }
+        };
+        return renderItem(
+            container,
+            this.getElementOptions(merge(options, { value }), callback)
+        );
     }
 
     private getElementOptions(
         options: MenuItem.Options,
-        callback?: () => void
+        callback?: Function
     ): MenuItem.Options {
 
         return {
             id: options.id,
             name: options.id,
             title: options.collapsable ? '' : options.text || '',
-            callback,
             item: this,
-            nestedOptions: options.nestedOptions,
             icon: options.icon,
             mousedown: options.events && options.events.onmousedown,
             value: options.value || '',
-            onchange: options.events && options.events.change,
+            onchange: callback,
```

Comment:
WIll not break renderInput where onChange is used? So when we're typing the "new chart's title", all is correct?